Rational-quadratic spline transformations

VasylHafych opened the pull request:
Hi @oschulz, @Micki-D,

Here is a draft of the Monotonic Rational-quadratic Splines used in the Neural Spline Flows [paper](https://proceedings.neurips.cc/paper/2019/hash/7ac71d433f282034e088473244df8c02-Abstract.html). Pullbacks for gradient evaluations are already implemented, but yet without `@atomic` operations. There is a demo notebook that allows playing with toy fits. 

I will keep this WIP branch here to get your comments. Also, the updates with atomic operations should be pushed here soon.

VasylHafych committed:
Added spline transformations with custom pullbacks

VasylHafych committed:
Improved demo notebook

VasylHafych committed:
Minor changes

VasylHafych committed:
Merge pull request #2 from bat/main

CompatHelper: bump compat for DocStringExtensions to 0.9, (keep exist…

VasylHafych committed:
Merge branch 'main' into splines-dev

oschulz commented:
Thanks!  @Micki-D want to start playing with this? @VasylHafych , should be wait for the atomic ops before merging?

VasylHafych commented:
Yes, let's merge and review the final version.

Micki-D commented:
Yes, I'm actually already working on the atomic addition.

________________________________
Von: Vasyl Hafych ***@***.***>
Gesendet: Dienstag, 28. Juni 2022 18:34:28
An: bat/EuclidianNormalizingFlows.jl
Cc: Dudkowiak, Michael; Mention
Betreff: Re: [bat/EuclidianNormalizingFlows.jl] Rational-quadratic spline transformations (PR #15)


Yes, let's merge and review the final version.

—
Reply to this email directly, view it on GitHub<https://github.com/bat/EuclidianNormalizingFlows.jl/pull/15#issuecomment-1168956410>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AUAIOCXACCLHQ657BHSMMS3VRMSRJANCNFSM52COYKPQ>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

VasylHafych committed:
Debug log

VasylHafych committed:
Updated pullback: Tangent multiplication without storing sample index

VasylHafych committed:
Merge branch 'pullback-optim' into splines-dev

Micki-D committed:
Add minor changes

Remove @Const macros

Fix array indexing in Kernels

Fix missing B arg

Fix bugs, seperate param and param grad evaluation

Fix example notebook

VasylHafych committed:
Merge pull request #7 from Micki-D/splines-dev

Pull out Array allocation from kernels
Separate parameter evaluation and parameter and parameter gradient evaluation

Micki-D committed:
Remove Loops from pb_kernel(), rename utils, represent LogJac as Matrix in Kernels

Add minor changes

VasylHafych committed:
Merge pull request #8 from Micki-D/splines-dev

Remove Loops from pb_kernel(), rename utils, represent LogJac as Matr…

VasylHafych committed:
Removed array allocations form kernels, cleaned

VasylHafych committed:
Added double type structure: TrainableRQSpline and RQSpline
